pkg/ingester/streams_map.go

```diff
@@ -0,0 +1,103 @@
+package ingester
+
+import (
+	"sync"
+
+	"github.com/prometheus/common/model"
+	"go.uber.org/atomic"
+)
+
+type streamsMap struct {
+	consistencyMtx sync.Mutex
```

Comment:
nit: Can you add a comment that this is for keeping the consistency between the two map please ?

Reply:
The comment commit was missing from this PR, I'll fill it up in next one.